### PR TITLE
Document the new implementation of `polymer lint`

### PR DESCRIPTION
**Pull request by rictic**
/cc @justinfagnani 

The new implementation will be released on Monday with the push of v0.18.0 to the `latest` npm tag.

### CODE REVIEW

File: app/2.0/docs/tools/polymer-cli.md [hunk 1]
```diff
@@ -303,20 +303,43 @@ Open up a page other than the default `index.html` in a specific browser
 
 ### `polymer lint` {#lint}
 
-Lints element(s)..
+Analyze your project for syntax errors, missing imports, bad databinding expressions and more.
```
Comment (arthurevans) [inline app/2.0/docs/tools/polymer-cli.md]: Is it worth adding a note here about what `polymer lint` _doesn't_ do? 

As I understand it, it doesn't catch general JavaScript issues, so people might want to supplement it with a general-purpose linter like ESLint or JSHint.

File: app/1.0/docs/tools/polymer-cli.md [hunk 2]
```diff
@@ -275,19 +275,44 @@ Open up a page other than the default `index.html` in a specific browser
...
+The output of `polymer lint` looks like this:
+
+```
+            <iron-collapse>
+            ~~~~~~~~~~~~~~~
+
+index.html(83,12) warning [undefined-elements] - The element iron-collapse is not defined
+```
+
+This means that on line 83 of `index.html` there's an `<iron-collapse>` tag, but the linter can't find the definition of the `iron-collapse` custom element. This probably means that there's a missing HTML import in `index.html`, but if we wanted to ignore this warning, we'd add `"undefined-elements"` to the `"ignoreWarnings"` array in `polymer.json`.
```
Comment (arthurevans) [inline app/1.0/docs/tools/polymer-cli.md]: Suggest breaking the last sentence and writing in 2nd person for consistency:

in `index.html`. To ignore this warning, add `"undefined-elements"` to the `"ignoreWarnings"` array in `polymer.json`.

**Comment by arthurevans** [comment]
A few small suggestions otherwise LGTM.

File: app/1.0/docs/tools/polymer-cli.md (hunk 2 above)
Comment (rictic) [inline app/1.0/docs/tools/polymer-cli.md]: Done!

File: app/2.0/docs/tools/polymer-cli.md (hunk 1 above)
Comment (rictic) [inline app/2.0/docs/tools/polymer-cli.md]: +1, added some more text for clarity

**Comment by rictic** [comment]
Comments addressed, please take a look!

**Pull request merged**